src/System.Text.Json/tests/JsonNodeTests.JsonNumber.cs

```diff
@@ -0,0 +1,534 @@
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Text.Json.Tests
+{
+    public static class JsonNumberTests
+    {
+        private delegate bool TryGetValue<T>(JsonNumber number, out T result);
+
+        private static void TestInitialization<T>(
+                T value,
+                Func<T, JsonNumber> ctor,
+                Action<JsonNumber, T> setter,
+                Func<JsonNumber, T> getter, 
+                TryGetValue<T> tryGetter,
+                Func<T, JsonNumber> implicitCaster)
+        {
+            // Default constructor:
+            JsonNumber number = new JsonNumber();
+            setter(number, value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // Numeric type constructor:
+            number = ctor(value);
+            AssertValue(value, number, getter, tryGetter);
+
+            // String constructor:
+            number = new JsonNumber(value.ToString());
+            AssertValue(value, number, getter, tryGetter);
+            
+            // Implicit cast:
+            number = implicitCaster(value);
+            AssertValue(value, number, getter, tryGetter);
+        }
+
+        private static void AssertValue<T>(
+                T value,
+                JsonNumber number,
+                Func<JsonNumber, T> getter,
+                TryGetValue<T> tryGetter)
+        {
+            Assert.Equal(value, getter(number));
+            Assert.True(tryGetter(number, out T result));
+            Assert.Equal(value, result);
+        }
+
+        [Fact]
+        public static void TestDefaultCtor()
+        {
+            var jsonNumber = new JsonNumber();
+            Assert.Equal(0, jsonNumber.GetByte());
+            Assert.Equal(0, jsonNumber.GetInt16());
+            Assert.Equal(0, jsonNumber.GetInt32());
+            Assert.Equal(0, jsonNumber.GetInt64());
+            Assert.Equal(0, jsonNumber.GetSingle());
+            Assert.Equal(0, jsonNumber.GetDouble());
+            Assert.Equal(0, jsonNumber.GetSByte());
+            Assert.Equal((ushort)0, jsonNumber.GetUInt16());
+            Assert.Equal((uint)0, jsonNumber.GetUInt32());
+            Assert.Equal((ulong)0, jsonNumber.GetUInt64());
+            Assert.Equal(0, jsonNumber.GetDecimal());
+        }
+
+        [Theory]
+        [InlineData(17)]
+        [InlineData(byte.MinValue)]
+        [InlineData(byte.MaxValue)]
+        public static void TestByte(byte value)
+        {
+            TestInitialization(
+                value,
+                v => new JsonNumber(v),
+                (number, v) => number.SetByte(v),
+                number => number.GetByte(),
+                (JsonNumber number, out byte v) => number.TryGetByte(out v),
+                v => v);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-17)]
+        [InlineData(17)]
+        [InlineData(byte.MinValue-1)]
```

Comment:
nit: spacing (same below in other tests)
```suggestion
        [InlineData(byte.MinValue - 1)]
```